### PR TITLE
Embedded LXD Profile

### PR DESCRIPTION
**Pull request by VariableDeclared**
Added a LXD profile to the Kubernetes-worker charm to LXD deployments.

[LP# 1835078](https://bugs.launchpad.net/charm-kubernetes-master/+bug/1835078)
---
[Related PR](https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/24)

### CODE REVIEW

**Comment by VariableDeclared** [comment]
Testing stripped profile. Do not merge.

File: README.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ firewall rules may not be able to contact the Charm Store. In these network
 restricted  environments the resource can be uploaded to the model by the Juju
 operator.
 
+## Deploying on LXD
+
+When deploying on LXD this charm will deploy its own profile, this profile can be inspected by running `lxd profile list` and choosing the profile matching your juju model name, once you have this you can run `lxc profile show ${PROFILE_NAME}`.
```
Comment (hyperbolic2346) [inline README.md]: Is juju going to rename this profile to match the model name?

File: lxd-profile.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    lxc.mount.auto=proc:rw sys:rw
+    lxc.cap.drop=
+  security.nesting: true
+  security.privileged: true
+description: ""
+devices:
+  aadisable:
+    path: /dev/kmsg
+    source: /dev/kmsg
+    type: unix-char
```
Comment (hyperbolic2346) [inline lxd-profile.yaml]: We don't need to match the profile [from the wiki](https://github.com/charmed-kubernetes/bundle/wiki/Deploying-on-LXD#the-profile)? We're missing aadisable1, aadisable2, and boot.autostart.

Comment (VariableDeclared) [reply lxd-profile.yaml]: Nope, so these keys in question were failing to validate with juju even when being [deployed on a non-lxd cloud](https://bugs.launchpad.net/juju/+bug/1836226). So they were removed, and the stripped down profile was tested againsted the validation tests in the Jenkins repo. :smile:

File: README.md (hunk 1 above)
Comment (VariableDeclared) [reply README.md]: Yes, I will clear up docs to clarify this

File: README.md [hunk 3]
```diff
@@ -40,6 +40,12 @@ firewall rules may not be able to contact the Charm Store. In these network
 restricted  environments the resource can be uploaded to the model by the Juju
 operator.
 
+## Deploying on LXD
+
+When deploying on LXD juju will deploy the lxd profile found inside of this charm, renaming it in the process this profile can be inspected by running `lxd profile list` and choosing the profile matching your juju model name, once you have this you can run `lxc profile show ${PROFILE_NAME}`.
```
Comment (kwmonroe) [inline README.md]: @VariableDeclared, there's no prize for longest sentence ;)

That aside, I'm not sure charm operators need to know this level of detail about the embedded profile. @evilnick, do we have any doc blurbs about cdk on lxd that we could link from here?

Comment (evilnick) [reply README.md]: @kwmonroe No. all the localhost stuff is currently predicated around conjure-up. There won't be any lxd docs until I write them based on however this turns out.

Comment (VariableDeclared) [reply README.md]: I had a suspicion this might be the case. I can cut down the paragraph if required :smile:

**Pull request merged**